src/test/ui/type-alias-impl-trait/multiple-def-uses-in-one-fn.stderr

```diff
@@ -0,0 +1,35 @@
+error: defining use generics `[B, A]` differ from previous defining use
+  --> $DIR/multiple-def-uses-in-one-fn.rs:10:45
+   |
+LL | fn f<A, B: 'static>(a: &'static A, b: B) -> (X<A, B>, X<B, A>) {
+   |                                             ^^^^^^^^^^^^^^^^^^
+   |
+note: previous defining use with different generics `[A, B]` found here
+  --> $DIR/multiple-def-uses-in-one-fn.rs:10:45
+   |
+LL | fn f<A, B: 'static>(a: &'static A, b: B) -> (X<A, B>, X<B, A>) {
+   |                                             ^^^^^^^^^^^^^^^^^^
+
+error[E0277]: the trait bound `&'static B: From<&A>` is not satisfied
+  --> $DIR/multiple-def-uses-in-one-fn.rs:7:16
+   |
+LL | type X<A, B> = impl Into<&'static A>;
+   |                ^^^^^^^^^^^^^^^^^^^^^ the trait `From<&A>` is not implemented for `&'static B`
+   |
+   = note: required because of the requirements on the impl of `Into<&'static B>` for `&A`
+
```

Comment:
this error will go away if we use the error type for known-broken TAITs

Reply:
So if we use `TyKind::Err` we would end up with _no_ error that can potentially help a user with the same confusion I'm having, right?

Reply:
So, in my opinion, we probably *don't* want this, right?

In a perfect world, this would be be some sort of error like "the trait bound is not satisfied, because of the types used in it's defining use." But when we have multiple defining uses, it's just arbitrary what bounds we're checking. And I feel like it just adds more info that might not even be relevant.

Reply:
Yea, that's my point, too. Adding the relevant trait bounds does not actually fix the issue (see the `multiple-def-uses-in-one-fn2.rs` test, which does just that).